src/contracts/strategies/StrategyFactory.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
+import "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
+import "./StrategyFactoryStorage.sol";
+import "./StrategyBase.sol";
+import "../permissions/Pausable.sol";
+
+/**
+ * @title Factory contract for deploying BeaconProxies of a Strategy contract implementation for arbitrary ERC20 tokens
+ *        and automatically adding them to the StrategyWhitelist in EigenLayer.
+ * @author Layr Labs, Inc.
+ * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
+ * @dev This may not be compatible with non-standard ERC20 tokens. Caution is warranted.
+ */
+contract StrategyFactory is StrategyFactoryStorage, OwnableUpgradeable, Pausable {
+    uint8 internal constant PAUSED_NEW_STRATEGIES = 0;
+
+    /// @notice EigenLayer's StrategyManager contract
+    IStrategyManager public immutable strategyManager;
+
+    /// @notice Since this contract is designed to be initializable, the constructor simply sets the immutable variables.
+    constructor(IStrategyManager _strategyManager) {
+        strategyManager = _strategyManager;
+        _disableInitializers();
+    }
+
+    function initialize(
+        address _initialOwner,
+        IPauserRegistry _pauserRegistry,
+        uint256 _initialPausedStatus,
+        IBeacon _strategyBeacon
+    ) public virtual initializer {
+        _transferOwnership(_initialOwner);
+        _initializePauser(_pauserRegistry, _initialPausedStatus);
+        _setStrategyBeacon(_strategyBeacon);
+    }
+
+    /**
+     * @notice Deploy a new strategyBeacon contract for the ERC20 token.
+     * @dev A strategy contract must not yet exist for the token.
+     * $dev Immense caution is warranted for non-standard ERC20 tokens, particularly "reentrant" tokens
+     * like those that conform to ERC777.
+     */
+    function deployNewStrategy(IERC20 token)
```

Comment:
I could not find an event emitted when a new strategy gets deployed. EigenPodManager, for example, emits a `PodDeployed` event whenever an EigenPod is created, which can be used in [subgraph templates](https://thegraph.com/docs/en/developing/creating-a-subgraph/#data-source-templates) to index instantiated contracts. Is it possible to do the same here?

Reply:
We can use the `StrategyAddedToDepositWhitelist` event in the StrategyManager instead? Or do you want to track the count of contracts deployed from this contract?

Reply:
`StrategyAddedToDepositWhitelist` tracks only EigenLayer approved strategies that are shown on the UI, right?

Reply:
When we deploy a strategy through this method, it is added to the deposit whitelist. So it would be both EigenLayer approved AND permissionless strategies